ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs

```diff
@@ -368,14 +409,37 @@ trimSnapshots tracer hasFS DiskPolicy{..} = do
   Internal: reading from disk
 -------------------------------------------------------------------------------}
 
--- | On disk snapshots are numbered monotonically
-newtype DiskSnapshot = DiskSnapshot Int
+data DiskSnapshot = DiskSnapshot {
+      -- | Snapshots are numbered. We will try the snapshots with the highest
+      -- number first.
+      --
+      -- When creating a snapshot, we use the slot number the ledger state it
+      -- corresponds to as the snapshot number. This gives an indication of how
+      -- recent the snapshot is.
+      --
+      -- Note that the snapshot names are only indicative, we don't rely on the
+      -- snapshot number matching the slot number of the corresponding ledger
+      -- state. We only use the snapshots numbers to determine the order in
+      -- which we try them. If a user chooses to rename their snapshots, we'll
```

Comment:
Delete the sentence starting with "If ..", we already assume the user is not messing with the database.